src/job/compactor.rs

```diff
@@ -258,3 +260,22 @@ async fn run_clean_done_jobs() -> Result<(), anyhow::Error> {
         tokio::time::sleep(tokio::time::Duration::from_secs(time as u64)).await;
     }
 }
+
+async fn run_enrichment_table_merge() -> Result<(), anyhow::Error> {
+    log::info!("[COMPACTOR::JOB] Running enrichment table merge");
+    let Ok(locker) = infra::dist_lock::lock("/enrichment_table/merge", 0).await else {
+        log::error!("[COMPACTOR::JOB] Failed to acquire lock for enrichment table merge");
+        return Ok(());
+    };
+
+    let handle = tokio::task::spawn(async move {
+        if let Err(e) = crate::service::enrichment::storage::s3::run_merge_job().await {
+            log::error!("[COMPACTOR::JOB] run enrichment table merge error: {e}");
+        }
+    });
+    handle.await.unwrap();
+    if let Err(e) = infra::dist_lock::unlock(&locker).await {
+        log::error!("[COMPACTOR::JOB] Failed to release lock for enrichment table merge: {e}");
+    }
+    Ok(())
```

Comment:
the lock can't use like this. any lock it have a timeout, `0` means default, not forever. there is no forever lock.